monitoring/api/v3/alerts-client/snippets_test.py

```diff
@@ -126,6 +131,12 @@ def invoke_sample():
         snippets.replace_notification_channels(
             pochan.project_name, alert_policy_id, [notification_channel_id])
 
+    # This sleep call is for mitigating the following error:
+    # "409 Too many concurrent edits to the project configuration.
+    # Please try again."
+    # Having multiple projects will void this `sleep()` call.
+    # See also #3310
+    time.sleep(2)
```

Comment:
Do we still need the sleep if we are only calling the snippet once? 

How does the sleep differ from the @retry annotation on `invoke_sample`? Would it be better to just retry the entire test?

Reply:
> Do we still need the sleep if we are only calling the snippet once?
I think it's still safe to have sleep because if we have n builds simultaneously, there will be 2*n api calls in a short period of time.

Retrying entire test sounds good. Maybe I can mark them as flaky with some delay.

Reply:
Flakey can do something like this:

```python
def delay_on_exception(err, *args):
    if issubclass(err[0], ProductCrashedError):
       time.sleep(1)
       return True # will retry
    return False

@flaky(rerun_filter=delay_on_exception)
def test_something_else():
```